pkg/compliance/checks/env/env.go

```diff
@@ -0,0 +1,21 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2020 Datadog, Inc.
+
+package env
+
+import "github.com/DataDog/datadog-agent/pkg/compliance"
+
+// Env provides an abstraction for various environment methods needed by checks
+type Env interface {
```

Comment:
I feel like this a mix of multiple objects. It makes sense to have this interface, however I see three objects there:
- Accessing clients (we can extend that to reporter)
- Accessing shared configuration (hostname, etc..)
- Some utils
- Resolver

Currently the builder implements everything, which is practical, but maybe not the cleanest separation of concerns.

I believe we should keep this object to access Clients and Configuration (though we could create a `Configuration` interface and have a method in this `Env` interface to get it).
It will be useful when we'll need to implement (probably pretty soon) some customisation on top of our static benchmark YAML (for instance, Kubernetes base path might not be `/etc/kubernetes/`, or nodeRole for etcd checks might not be `etcd`, etc.).

I'd also split off the resolver as it's a different logic and give a resolver object to the checks. We may be able to put some utils in there too (normalizing path is kind of resolving something).

We could rename env as `CheckContext` or some sort of `Context`, being an entry point to all this.

Reply:
I like the `Context` name! Let me think about splitting, I will add a comment here once I have some ideas on that.

Reply:
What do you think about this split as the first step? While a subtle change it's a but more structured?

```
type Context interface {
	Clients
	Configuration
	Reporter() compliance.Reporter
}

type Clients interface {
	DockerClient() DockerClient
	AuditClient() AuditClient
	KubeClient() KubeClient
}

type Configuration interface {
	Hostname() string
	EtcGroupPath() string
	NormalizePath(path string) string
	ResolveValueFrom(valueFrom compliance.ValueFrom) (string, error)
}
```